projects/blindex/index.js

```diff
@@ -107,15 +186,107 @@ async function uniswapV2Tvl(timestamp, ethblock, chainBlocks) {
   return balances;
 }
 
+async function getAllBDStables(block, bdxTokenAddress, chainName) {
+  const bdStables = [];
+  const bdstablesLength = (
+    await sdk.api.abi.call({
+      target: formatAddressChecksum(bdxTokenAddress, chainName),
+      abi: abi["getBdStablesLength"],
+      chain: chainName,
+      block,
+    })
+  ).output;
+
+  for (let index = 0; index < bdstablesLength; index++) {
+    bdStables.push({
+      address: (
+        await sdk.api.abi.call({
+          target: formatAddressChecksum(bdxTokenAddress, chainName),
+          abi: abi["getBDStable"],
+          chain: chainName,
+          block,
+          params: index,
+        })
+      ).output,
+    });
+  }
+
+  return bdStables;
+}
+
+async function getBdxPriceInUSD(block, chainName) {
+  const bdusAddress = formatAddressChecksum(
+    "0xb450ff06d950efa9a9c0ad63790c51971c1be885",
+    chainName
+  );
+
+  const bdxPriceInUsd_d12 = (
+    await sdk.api.abi.call({
+      target: bdusAddress,
+      abi: abi["getBDXPriceUsdD12"],
+      chain: chainName,
+      block,
+    })
+  ).output;
+
+  return bdxPriceInUsd_d12 / 10 ** 12;
+}
+
+// TODO: This is needed until BDX will be avilable on Coingecko
+async function convertBdxToUsdc(block, chainName, balances) {
+  const bdxTokenAddress = chains[chainName].bdxTokenAddress;
+  const coingeckoMapBdxAddress = mapCoingeckoAddress(
+    chainName,
+    bdxTokenAddress
+  );
+
+  balances["usd-coin"] +=
+    balances[coingeckoMapBdxAddress] *
+    (await getBdxPriceInUSD(block, chainName));
+  balances[coingeckoMapBdxAddress] = 0;
+
+  return balances;
+}
+
+async function tvl(chainName, block) {
+  const balancesArray = [];
+
+  //=======
+  // AMM
+  //=======
+  balancesArray.push(await uniswapV2Tvl(block, chainName));
+
+  //===================
+  // Collateral
+  //===================
+  const bdstables = await getAllBDStables(
+    block,
+    chains[chainName].bdxTokenAddress,
+    chainName
+  );
+  for (let index = 0; index < bdstables.length; index++) {
+    balancesArray.push(
+      await getBDStableCollateralBalances(block, chainName, bdstables[index])
+    );
+  }
 
+  const balances = sumBalances(balancesArray);
 
+  // TODO: This should be removed when BDX will be listed on Coingecko
+  return await convertBdxToUsdc(block, chainName, balances);
+}
+
+const rsk = async function rskTvl(timestamp, ethBlock, chainblocks) {
+  const block = await getBlock(timestamp, "rsk", chainblocks);
+  return tvl("rsk", block);
+};
 
 module.exports = {
   misrepresentedTokens: true,
   methodology:
     "(1) AMM LP pairs - All the liquidity pools from the Factory address are used to find the LP pairs. (2) Collateral - All the collateral being used to support the stable coins - Bitcoin, Ethereum & BDX",
   rsk: {
-    tvl: sdk.util.sumChainTvls([uniswapV2Tvl,collaterlTvl]),
+    tvl: rsk,
   },
-  
+  tvl: sdk.util.sumChainTvls([rsk]),
```

Comment:
this line can be removed, this step is no longer required

Reply:
So when going multichain it will know how to handle it fine by just having more chains in the object?

Reply:
yup, example: https://github.com/Skyge/DefiLlama-Adapters/blob/04d48c2bf419661e8f9991f6408eb756358bedae/projects/dforce/index.js